google-cloud-storage/src/test/java/com/google/cloud/storage/ZOptTest.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.storage;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.fail;
+
+import com.google.cloud.storage.spi.v1.StorageRpc;
+import com.google.common.collect.ImmutableMap;
+import java.util.Collections;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+import org.junit.Test;
+
+public final class ZOptTest {
+
+  @Test
+  public void happyPath() {
```

Comment:
What do you think about adding a test to check that all StorageRpc.Option.* values are handled by ZOpt? 

It will help when we introduce new options in the future; It's probably going to be a "duh" moment, but it will be helpful to make sure there are no orphens introduced.